src/main/scala/mesosphere/marathon/core/appinfo/impl/DefaultAppInfoService.scala

```diff
@@ -0,0 +1,50 @@
+package mesosphere.marathon.core.appinfo.impl
+
+import mesosphere.marathon.core.appinfo.AppInfo.Embed
+import mesosphere.marathon.core.appinfo.{ AppInfo, AppSelector, AppInfoService }
+import mesosphere.marathon.state.{ AppDefinition, GroupManager, GroupRepository, AppRepository, PathId }
+
+import scala.concurrent.Future
+
+private[appinfo] class DefaultAppInfoService(
+    groupManager: GroupManager,
+    appRepository: AppRepository,
+    newBaseData: () => AppInfoBaseData) extends AppInfoService {
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  override def queryForAppId(id: PathId, embed: Set[Embed]): Future[Option[AppInfo]] = {
+    appRepository.currentVersion(id).flatMap {
```

Comment:
As long as there is a difference between groupRepo and appRepo, I would use the groupManager to fetch the app.